internal/commands/relationship.go

```diff
@@ -209,60 +210,79 @@ func buildReadRequest(cmd *cobra.Command, args []string) (*v1.ReadRelationshipsR
 		}
 	}
 
+	limit := cobrautil.MustGetUint32(cmd, "page-limit")
 	return &v1.ReadRelationshipsRequest{
 		RelationshipFilter: readFilter,
+		OptionalLimit:      limit,
 	}, nil
 }
 
 func readRelationships(cmd *cobra.Command, args []string) error {
-	request, err := buildReadRequest(cmd, args)
-	if err != nil {
-		return err
-	}
-
-	request.Consistency, err = consistencyFromCmd(cmd)
+	spicedbClient, err := client.NewClient(cmd)
 	if err != nil {
 		return err
 	}
 
-	spicedbClient, err := client.NewClient(cmd)
+	request, err := buildReadRequest(cmd, args)
 	if err != nil {
 		return err
 	}
+	limit := request.OptionalLimit
 
-	log.Trace().Interface("request", request).Msg("reading relationships")
-	resp, err := spicedbClient.ReadRelationships(cmd.Context(), request)
+	request.Consistency, err = consistencyFromCmd(cmd)
 	if err != nil {
 		return err
 	}
 
+	lastCursor := request.OptionalCursor
 	for {
-		if err := cmd.Context().Err(); err != nil {
-			return err
-		}
-
-		msg, err := resp.Recv()
-		if errors.Is(err, io.EOF) {
-			return nil
-		}
-
+		request.OptionalCursor = lastCursor
+		log.Trace().Interface("request", request).Msg("reading relationships page")
+		readRelClient, err := spicedbClient.ReadRelationships(cmd.Context(), request)
 		if err != nil {
 			return err
 		}
 
-		if cobrautil.MustGetBool(cmd, "json") {
-			prettyProto, err := PrettyProto(msg)
-			if err != nil {
+		relCount := uint32(0)
+		for {
+			if err := cmd.Context().Err(); err != nil {
 				return err
 			}
 
-			console.Println(string(prettyProto))
-		} else {
-			relString, err := relationshipToString(msg.Relationship)
+			msg, err := readRelClient.Recv()
+			if errors.Is(err, io.EOF) {
+				break
+			}
+
 			if err != nil {
 				return err
 			}
-			console.Println(relString)
+
+			lastCursor = msg.AfterResultCursor
+			relCount++
+			if cobrautil.MustGetBool(cmd, "json") {
+				prettyProto, err := PrettyProto(msg)
+				if err != nil {
+					return err
+				}
+
+				console.Println(string(prettyProto))
+			} else {
+				relString, err := relationshipToString(msg.Relationship)
+				if err != nil {
+					return err
+				}
+				console.Println(relString)
+			}
```

Comment:
This seems like a good thing to pull out into it's own function.
I was a bit confused by the `relCount++` then a conditional, but I wouldn't have been if i saw `relCount++; printRel(...)`